### PR TITLE
Split parsing from graph building.

### PR DESCRIPTION
**Pull request by dnesbitt61**
Added PBFParser class. Split apart OSMNode and Node.

### CODE REVIEW

File: src/mjolnir/pbfgraphbuilder.cc [hunk 1]
```diff
@@ -87,15 +88,49 @@ bool ParseArguments(int argc, char *argv[]) {
...
+  parser.PostProcess();
+  return parser.edge_count();
+}
+
+/**
  * Build local graph from protocol buffer input.
  */
 void BuildLocalGraphFromPBF(const boost::property_tree::ptree& pt,
                const std::vector<std::string>& input_files) {
+  // Stores all the ways that are part of the road network
```
Comment (kevinkreiser) [inline src/mjolnir/pbfgraphbuilder.cc]: can we wrap all of thse in a struct defined within PBFParser, and have  the parser return a const ref to the struct which we can then pass along to graphbuilder::build?

**Comment by kevinkreiser** [comment]
looks good other than comment above

**Comment by dgearhart** [comment]
:+1:

**Comment by gknisely** [comment]
:ship:

**Pull request merged**